agent/taskresource/credentialspec/credentialspec_windows.go

```diff
@@ -298,15 +297,12 @@ func (cs *CredentialSpecResource) GetName() string {
 // Create is used to create all the credentialspec resources for a given task
 func (cs *CredentialSpecResource) Create() error {
 	var err error
-	// To fail fast, check execution role first
+	var iamCredentials credentials.IAMRoleCredentials
```

Comment:
Sorry. I should say it more clearly. Since here is not a pointer, you should check whether it is an empty struct or not. You don't need to declare the variable at the beginning. 
Just do 
```
executionCredentials, ok := cs.credentialsManager.GetTaskCredentials(cs.getExecutionCredentialsID())
```
and in the handleS3CredentialspecFile/handleSSMCredentialspecFile,  check whether it is an empty struct, something like
```
iamCredentials ==  credentials.IAMRoleCredentials{}
```

Reply:
Instead of copying the credentials, I'm passing the credentials around via reference.

Reply:
Thanks for explaining. Fixed.